sql/src/planner.rs

```diff
@@ -397,49 +417,39 @@ impl<'a, P: MetaProvider> PlannerDelegate<'a, P> {
             )
         }
 
-        // Build primary key, the builder will check timestamp column is in primary key.
-        if let Some(idx) = primary_key_constraint_idx {
-            // If primary key is already provided, use that primary key.
-            if let TableConstraint::Unique { columns, .. } = &stmt.constraints[idx] {
-                for col in columns {
-                    let key_column = if TSID_COLUMN == col.value {
-                        schema_builder = schema_builder.enable_tsid_primary_key(true);
-                        Self::tsid_column_schema()?
-                    } else {
-                        name_column_map
-                            .remove(&*col.value)
-                            .with_context(|| PrimaryKeyNotFound {
-                                name: col.value.clone(),
-                            })?
-                    };
-                    // The schema builder will checks there is only one timestamp column in primary
-                    // key.
+        let timestamp_col_idx = timestamp_column_idx.context(RequireTimestamp)?;
+        // The key columns have been consumed.
```

Comment:
Is this still right?
Loop below still use key columns

Reply:
I need to keep the order of the columns in the loop, so I have to loop through the columns in the `stmt.column`.
the `timestamp_col_idx `  is evaluated in the loop against the column index and a timestamp key is added.